### PR TITLE
salesforce source

### PR DESCRIPTION
**Pull request by cgardens**
## What
* salesforce singer

## How
*Describe the solution*

## Checklist
- [ ] Run standard tests as part of build.
- [ ] Check how enum works in the ui

### CODE REVIEW

File: airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/base/TestSource.java [hunk 1]
```diff
@@ -186,7 +186,6 @@ public void testDiscover() throws Exception {
     // the worker validates that it is a valid catalog, so we do not need to validate again (as long as
     // we use the worker, which we will not want to do long term).
     assertNotNull(output.getOutput().get().getCatalog());
-    assertEquals(getCatalog(), output.getOutput().get().getCatalog());
```
Comment (cgardens) [inline airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/base/TestSource.java:189]: i removed this for now. right now that catalog is used for this comparison and the catalog used to configure the run tests. the problem here is that for the catalog to be used in the comparison in this test, we need to pass in the full catalog (which for salesforce is gigantic). for now, i prefer to pass in the smaller one that is a subset for to use in the run command. we can add a separate argument to pass in the full catalog and then bring this assertion back.

File: .github/workflows/gradle.yml [hunk 2]
```diff
@@ -53,6 +54,9 @@ jobs:
       - name: Run Integration Tests
         run: ./gradlew --no-daemon integrationTest --scan
 
+      - name: Run Standard Source Tests
```
Comment (cgardens) [inline .github/workflows/gradle.yml:57]: adding standard source tests to build as these are the only tests that currently run for salesforce and github.

File: airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/base/TestSource.java (hunk 1 above)
Comment (sherifnada) [reply airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/base/TestSource.java:189]: Why not pass in the gigantic catalog?

**Review by sherifnada — APPROVED** [review]

File: airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/base/TestSource.java (hunk 1 above)
Comment (cgardens) [reply airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/base/TestSource.java:189]: >  i prefer to pass in the smaller one that is a subset for to use in the run command.
i only want to sync one stream, not all fifty thousand in the read test.

Comment (cgardens) [reply airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/base/TestSource.java:189]: also the schema is dynamically generated based on what's in salesforce. so the test will fail if anyone ever adds a field in that test account. in short it just adds alot of entropy for little gain.

**Pull request merged**